src/researches/buildKeywordForms.js

```diff
@@ -41,67 +42,97 @@ const getVariationsApostropheInArray = function( forms ) {
  *
- * @param {Object} paper The paper to analyze keyword forms for.
+ * @param {string} keyphrase The keyphrase of the paper (or a synonym phrase) to get forms for.
+ * @param {string} language The language to use for morphological analyzer and for function words.
+ * @param {boolean} morphologyRequired Whether the morphological analysis is required (i.e, Premium vs. Free).
  *
  * @returns {Array} Array of all forms to be searched for keyword-based assessments.
  */
-module.exports = function( paper ) {
-	let keyword = paper.getKeyword();
-	const language = getLanguage( paper.getLocale() );
-	const getForms = getFormsForLanguage[ language ];
-
-	/*
-	 * If the language is not yet supported with respect to morphological analysis, return keyword itself,
-	 * including variations of the apostrophe in case it is present.
-	 */
-	if ( isUndefined( getForms ) ) {
-		return unique( [].concat( keyword, getVariationsApostrophe( keyword ) ) );
+const buildForms = function( keyphrase, language, morphologyRequired = false ) {
+	if ( isUndefined( keyphrase ) || keyphrase === "" ) {
+		return [];
 	}
 
+	// If the keyphrase is embedded in double quotation marks, return keyword itself, without outer-most quotation marks.
 	const doubleQuotes = [ "“", "”", "〝", "〞", "〟", "‟", "„", "\"" ];
-	// If the keyword is embedded in double quotation marks, return keyword itself, without outer-most quotation marks.
-	if ( includes( doubleQuotes, keyword[ 0 ] ) && includes( doubleQuotes, keyword[ keyword.length - 1 ] ) ) {
-		keyword = keyword.substring( 1, keyword.length - 1 );
-		return unique( [].concat( keyword, getVariationsApostrophe( keyword ) ) );
+	if ( includes( doubleQuotes, keyphrase[ 0 ] ) && includes( doubleQuotes, keyphrase[ keyphrase.length - 1 ] ) ) {
+		keyphrase = keyphrase.substring( 1, keyphrase.length - 1 );
+		return [ unique( [].concat( keyphrase, getVariationsApostrophe( keyphrase ) ) ) ];
 	}
 
-	let forms = [];
+	const words = filterFunctionWords( getWords( keyphrase ), language );
 
-	// If the keyword is a single word return all its possible forms.
-	if ( keyword.indexOf( " " ) === -1 ) {
-		const wordToLowerCase = escapeRegExp( keyword.toLocaleLowerCase() );
-		forms = flatten( forms.concat( getForms( wordToLowerCase ) ) );
-		forms = unique( flatten( forms.concat( getVariationsApostropheInArray( forms ) ) ) );
-		return forms;
-	}
-
-	// If the keyword contains multiple words first filter all function words out, then build all possible forms for remaining contents words.
-	const keyWords = getWords( keyword );
-	const functionWords = getFunctionWords[ language ].all;
+	const getForms = getFormsForLanguage[ language ];
+	let forms = [];
 
-	const keyWordsFiltered = filter( keyWords, function( word ) {
-		return ( ! includes( functionWords, word.trim().toLocaleLowerCase() ) );
-	} );
+	if ( morphologyRequired === false || isUndefined( getForms ) ) {
```

Comment:
Perhaps add as a comment line: `Only return the keyword and the keyword with apostrophe variations if morphological forms should not or cannot be built. Otherwise additionally returns the morphological forms.`